homeassistant/components/zwave_js/update.py

```diff
@@ -0,0 +1,199 @@
+        )
+
+        self._attr_installed_version = self._attr_latest_version = node.firmware_version
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return entity specific state attributes."""
+
+        return {
+            ATTR_AVAILABLE_FIRMWARE_UPDATES: [
+                firmware.version
+                for firmware in sorted(
+                    self.available_firmware_updates,
+                    key=lambda f: AwesomeVersion(f.version),
+                )
+            ]
+        }
+
+    async def async_update(self) -> None:
+        """Update the entity."""
+        self.available_firmware_updates = (
+            await self.driver.controller.async_get_available_firmware_updates(
+                self.node, API_KEY_FIRMWARE_UPDATE_SERVICE
+            )
+        )
+        self._async_process_updates(False)
+
+    @callback
+    def _async_process_updates(self, write_state: bool = True) -> None:
+        """Process updates."""
+        if self.available_firmware_updates:
+            self._latest_version_firmware = firmware = get_latest_firmware(
+                self.available_firmware_updates
+            )
+            self._attr_latest_version = firmware.version
+        else:
+            self._latest_version_firmware = None
+            self._attr_latest_version = self._attr_installed_version
+        if write_state:
+            self.async_write_ha_state()
+
+    async def async_release_notes(self) -> str | None:
+        """Get release notes."""
+        if self._latest_version_firmware is None:
+            return None
+        return self._latest_version_firmware.changelog
+
+    async def async_install(
+        self, version: str | None, backup: bool, **kwargs: Any
+    ) -> None:
+        """Install an update."""
+        if version is None and self._latest_version_firmware:
+            firmware = self._latest_version_firmware
+        else:
+            try:
+                firmware = next(
+                    firmware
+                    for firmware in self.available_firmware_updates
+                    if firmware.version == version
+                )
+            except StopIteration as err:
+                raise ValueError(f"Version {version} not found") from err
+        self._attr_in_progress = True
+        try:
+            for file in firmware.files:
+                await self.driver.controller.async_begin_ota_firmware_update(
+                    self.node, file
+                )
+        except BaseZwaveJSServerError as err:
+            raise HomeAssistantError(err) from err
+        else:
+            self._attr_installed_version = firmware.version
+            self.available_firmware_updates.remove(firmware)
+            self._async_process_updates()
+        finally:
+            self._attr_in_progress = False
+
+    async def async_poll_value(self, _: bool) -> None:
+        """Poll a value."""
+        LOGGER.error(
```

Comment:
Should we raise an error ?

Reply:
we don't anywhere else (node status sensor, ping button). If we change the behavior here we should change it there as well

Reply:
ok not for this PR.